tests/test_qiskit_device.py

```diff
@@ -13,6 +74,43 @@
 ]
 
 test_device_options = [{}, {"optimization_level": 3}, {"optimization_level": 1}]
+backend = MockedBackend()
+legacy_backend = MockedBackendLegacy()
+
+
+class TestSupportForV1andV2:
+    """Tests compatibility with BackendV1 and BackendV2"""
+
+    @pytest.mark.parametrize(
+        "backend",
+        [
+            legacy_backend,
+            backend,
+        ],
+    )
+    def test_v1_and_v2_mocked(self, backend):
+        """Test that device initializes with no error mocked"""
+        dev = qml.device("qiskit.remote", wires=10, backend=backend, use_primitives=True)
+
+
+    @pytest.mark.parametrize(
+            "backend",
+            [
+                FakeManila(),
+                FakeManilaV2(),
+            ]
+    )
+    def test_v1_and_v2_manila(self, backend):
+        """Test that device initializes with no error with V1 and V2 backends by Qiskit"""
+        dev = qml.device("qiskit.remote", wires=5, backend=backend, use_primitives=True)
+        
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=[0])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+        
+        circuit(np.pi/2)
```

Comment:
Optional: we may add an `assertion` to test the circuit results. Could be skipped if the outputs are noisy.

Reply:
Outputs are noisy here since we are simulating a real backend. I talked about this with @lillian542 but didn't think of a good way to test the circuit results (as far as I know Qiskit doesn't provide a BackendV1 that is noise-free).